auto-value-gson-extension/src/main/java/com/mapbox/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -960,6 +959,11 @@ private MethodSpec createReadMethod(ClassName className,
       }
     }
     if (unrecognisedJsonPropertiesContainer != null) {
+      readMethod.beginControlFlow("if (unrecognised == null)");
+      readMethod.addStatement("unrecognised = new $T()", mapOfSerializableJsonElements);
+      readMethod.addStatement("builder.$L(unrecognised)", unrecognisedJsonPropertiesContainer.methodName);
+      readMethod.endControlFlow();
```

Comment:
@LukasPaczos , I had to make unrecognised container nullable by default to support [this](https://github.com/mapbox/auto-value-gson/pull/2/files#diff-56f3908417d36285da76c84ad4e4fb8cbc8de6c94e0d6fcfe36a6a9716446b83R64) scenario.